packages/client/src/components/HTMLResourceRenderer.tsx

```diff
@@ -85,3 +102,21 @@ export const HTMLResourceRenderer = ({
 };
 
 HTMLResourceRenderer.displayName = 'HTMLResourceRenderer';
+
+function postResponse(
+  uiActionResult: UIActionResult,
+  event: MessageEvent,
+  type: (typeof InternalMessageType)[keyof typeof InternalMessageType],
+  payload?: Record<string, unknown>,
+) {
+  if (uiActionResult.messageId) {
+    event.source?.postMessage({
+      type,
+      payload: {
+        messageId: uiActionResult.messageId,
+        originalMessage: event.data,
```

Comment:
Why do you send the original data back to the consumer? They have a message ID, won't that be enough for them to do any record-keeping they need themselves?

Reply:
I though it can actually save the embedded code from needing to keep the messages data for some cases (for example if they only need to refer to it when a response is back)

Reply:
I agree that it feels redundant. I don't think we should encourage consumers to rely on the external source relaying both the request and the response. Large payloads may also become problematic.